kollect-core/src/main/kotlin/kollect/Kollect.kt

```diff
@@ -0,0 +1,45 @@
+package kollect
+
+import arrow.TC
+import arrow.core.Tuple2
+import arrow.data.NonEmptyList
+import arrow.free.Free
+import arrow.higherkind
+
+typealias Kollect<A> = Free<KollectOpHK, A>
+
+sealed class KollectError: Throwable() {
```

Comment:
This is fine for now, we can figure out a way to make it extensible later but we need a `NoStackTrace` interface that overrides `fillInStackTrace` to avoid that overhead that we are not using.

Reply:
:+1: